### PR TITLE
pools-updater: Support secure Tor connection to sync data with Github

### PR DESCRIPTION
**Pull request by ayanamidev**
Using Axios instead of native https module from node.js.

Also, I have added a retry in case of rate limit applied due to the use of proxy or maybe some minor connection problem.

### CODE REVIEW

**Comment by wiz** [comment]
@knorrium since you last worked on fetching resource files over Tor, did you want to review this PR

File: backend/src/tasks/pools-updater.ts [hunk 1]
```diff
@@ -105,35 +111,45 @@ class PoolsUpdater {
...
-    });
+      if (config.SOCKS5PROXY.USERNAME && config.SOCKS5PROXY.PASSWORD) {
+        socksOptions.username = config.SOCKS5PROXY.USERNAME;
+        socksOptions.password = config.SOCKS5PROXY.PASSWORD;
+      }
+
+      axiosOptions.httpsAgent = new SocksProxyAgent(socksOptions);
+    }
+
+    while(retry < 5) {
```
Comment (ayanamidev) [inline backend/src/tasks/pools-updater.ts:139]: Should we expose this value to `config.json`?

File: backend/src/tasks/pools-updater.ts [hunk 2]
```diff
@@ -105,35 +111,45 @@ class PoolsUpdater {
...
+        const data = await axios.get(path, axiosOptions);
+        if (data.statusText !== 'OK' || !data.data) {
+          throw new Error(`Could not fetch data from Github, Error: ${data.status}`);
+        }
+        return data.data;
+      } catch (e) {
+        logger.err('Could not connect to Github. Reason: '  + (e instanceof Error ? e.message : e));
+        retry++;
+      }
+      await setDelay();
```
Comment (ayanamidev) [inline backend/src/tasks/pools-updater.ts:150]: Maybe adding this value to config file as well, I think 1 second spacing for retry could not be enough with unstable connections like Tor network.

**Comment by ayanamidev** [comment]
I have added a retry loop so that the worker could retry in case of connection failure or failure with rate limits, could anyone take a look?

File: backend/src/tasks/pools-updater.ts (hunk 1 above)
Comment (wiz) [reply backend/src/tasks/pools-updater.ts:139]: Yeah that would be great

File: backend/src/tasks/pools-updater.ts (hunk 2 above)
Comment (wiz) [reply backend/src/tasks/pools-updater.ts:150]: Probably 10 seconds would be better since it's not urgent to get the updated data?

**Review by wiz — APPROVED** [review]
Partially Tested ACK @ v2.4.0-dev [[d1671c4f](https://github.com/mempool/mempool/commit/d1671c4f)]

Only tested over clearnet, so @knorrium will have to test over Tor when he starts the release testing on Umbrel target devices

**Pull request merged**